### PR TITLE
Adding CI/CD for rd-cpp

### PR DESCRIPTION
**Pull request by DecoyRS**
Github actions for RD CPP
* Build rd-cpp on win with vs 2019, vs 2017, on linux and mac - clang
* Download and cache Ninja project generator
* Cache submodules
* Build rd-cpp
* Pack artifacts in zip files
* Publish artifcats to Bintray
NB: will need to swap some vars in yaml to redirect to official JB bintray instead of my personal

### CODE REVIEW

**Comment by mfilippov** [comment]
@DecoyRS Why all checks are red?

File: .github/workflows/build-rd-cpp.yml [hunk 1]
```diff
@@ -0,0 +1,204 @@
...
+  pull_request:
+    branches: [ master ]
+
+env:
+  NINJA_VERSION: 1.9.0
+  BUILD_TYPE: Release
+  VERSION: 201.1
+  # Need to change these values to JetBrains account
+  # And set `RD_CPP_BINTRAY_KEY` for RD repo
+  BINTRAY_USER: alexander-pirogov
```
Comment (mfilippov) [inline .github/workflows/build-rd-cpp.yml:15]: It looks like forgotten debug value.

File: .github/workflows/build-rd-cpp.yml [hunk 2]
```diff
@@ -0,0 +1,204 @@
...
+    name: ${{ matrix.config.name }}
+    runs-on: ${{ matrix.config.os }}
+    strategy:
+      fail-fast: false
+      matrix:
+        config:
+        - {
+            name: "RD CPP Windows Latest MSVC", artifact: "RD-CPP-Windows-MSVC-2019.zip",
+            os: windows-latest,
+            cc: "cl", cxx: "cl",
```
Comment (mfilippov) [inline .github/workflows/build-rd-cpp.yml:31]: Is newline required after cc?

File: .github/workflows/build-rd-cpp.yml [hunk 3]
```diff
@@ -0,0 +1,204 @@
...
+        elseif ("${{ runner.os }}" STREQUAL "Linux")
+          set(ninja_suffix "linux.zip")
+        elseif ("${{ runner.os }}" STREQUAL "macOS")
+          set(ninja_suffix "mac.zip")
+        endif()
+
+        set(ninja_url "https://github.com/ninja-build/ninja/releases/download/v${ninja_version}/ninja-${ninja_suffix}")
+        file(DOWNLOAD "${ninja_url}" ./ninja.zip SHOW_PROGRESS)
+        execute_process(COMMAND
+                        ${CMAKE_COMMAND} -E tar xvf ./ninja.zip)
```
Comment (mfilippov) [inline .github/workflows/build-rd-cpp.yml:101]: Newline there looks strange. Please use single code style in the file.

File: .github/workflows/build-rd-cpp.yml [hunk 4]
```diff
@@ -0,0 +1,204 @@
...
+          set(ninja_suffix "win.zip")
+        elseif ("${{ runner.os }}" STREQUAL "Linux")
+          set(ninja_suffix "linux.zip")
+        elseif ("${{ runner.os }}" STREQUAL "macOS")
+          set(ninja_suffix "mac.zip")
+        endif()
+
+        set(ninja_url "https://github.com/ninja-build/ninja/releases/download/v${ninja_version}/ninja-${ninja_suffix}")
+        file(DOWNLOAD "${ninja_url}" ./ninja.zip SHOW_PROGRESS)
+        execute_process(COMMAND
```
Comment (mfilippov) [inline .github/workflows/build-rd-cpp.yml:100]: We don't check the command result here.

File: .github/workflows/build-rd-cpp.yml [hunk 5]
```diff
@@ -0,0 +1,204 @@
...
+      working-directory: ${{ env.working-dir }}
+      shell: cmake -P {0}
+      run: |
+        set(ENV{CC} ${{ matrix.config.cc }})
+        set(ENV{CXX} ${{ matrix.config.cxx }})
+        file(GLOB all_files "*")
+        message(${all_files})
+
+        if ("${{ runner.os }}" STREQUAL "Windows" AND NOT "x${{ matrix.config.environment_script }}" STREQUAL "x")
+          execute_process(
```
Comment (mfilippov) [inline .github/workflows/build-rd-cpp.yml:119]: We don't check the command result here.

File: .github/workflows/build-rd-cpp.yml [hunk 6]
```diff
@@ -0,0 +1,204 @@
...
+
+        execute_process(
+          COMMAND ${CMAKE_COMMAND} --build build --config ${{ env.BUILD_TYPE }}
+          RESULT_VARIABLE result
+        )
+        if (NOT result EQUAL 0)
+          message(FATAL_ERROR "Bad exit status")
+        endif()
+
+    - name: Install Strip
```
Comment (mfilippov) [inline .github/workflows/build-rd-cpp.yml:184]: Maybe we should do it before building.

File: .github/workflows/build-rd-cpp.yml (hunk 2 above)
Comment (DecoyRS) [reply .github/workflows/build-rd-cpp.yml:31]: yaml works just fine with this formatting.
Same on line 29, where `name` and `artifact` on the same line

File: .github/workflows/build-rd-cpp.yml (hunk 6 above)
Comment (DecoyRS) [reply .github/workflows/build-rd-cpp.yml:184]: We can't do `Install` before `Build` in terms of cmake.
Install means "put all artifacts in dedicated places`.

**Comment by DecoyRS** [comment]
@mfilippov all checks are red, because rd-cpp is actually broken for Ninja build in `master` branch) I'm working on pull-request from ue4-adapt to master, will finish today.

File: .github/workflows/build-rd-cpp.yml (hunk 1 above)
Comment (DecoyRS) [reply .github/workflows/build-rd-cpp.yml:15]: I've deliberately left these values untill I'll have an official creds for JetBrains bintray account.

Comment (mfilippov) [reply .github/workflows/build-rd-cpp.yml:15]: I think we need to wait for it before merging.

File: .github/workflows/build-rd-cpp.yml (hunk 2 above)
Comment (mfilippov) [reply .github/workflows/build-rd-cpp.yml:31]: ok

File: .github/workflows/build-rd-cpp.yml (hunk 6 above)
Comment (mfilippov) [reply .github/workflows/build-rd-cpp.yml:184]: ok thanks for info

**Comment by mfilippov** [comment]
@DecoyRS Ok, I want to wait for green status before the merge.

File: .github/workflows/build-rd-cpp.yml [hunk 7]
```diff
@@ -0,0 +1,204 @@
+name: Build RD CPP
```
Comment (ulex) [inline .github/workflows/build-rd-cpp.yml:1]: Sync name with rd-net part please

File names also not in sync. I named rd-net build as `build.yaml` as I expected configuration for a different parts will be located in the same file. As we have `rd-cpp.yml` it seems logical to rename `build.yml` to `build-rd-net.yml`

**Pull request closed without merging**